wasm/validator/src/lib.rs

```diff
@@ -55,6 +54,19 @@ macro_rules! deny {
     };
 }
 
+/// Like `?` but uses [`deny`] macro instead of `Err`.
+#[macro_export]
+macro_rules! propagate {
```

Comment:
I would inline this to keep it simple. It just confuses users and makes us have to explain one more thing in an API that is complicated as it is

Reply:
> This macro is used in 13 places

This metric should be taken with a grain of salt. If I wrote a macro like this and used in in 100 places:
```rs
macro_rules! ret {
    ( $val:ident ) => {
        return $val
    }
}

fn return_val() -> usize {
    let a = 12;
    ret!(a);
}
```

would you say this is a useful macro? IMO this is what `propagate` is doing. It replaces a perfectly readable Rust syntax. It barely reduces the amount of repetition in the code. Not every code repetition is bad, especially if the amount of code is small. A macro should have some meat in it

Maybe someone else has an opinion?

Reply:
Ok, most of the time it's

```rust
if propagate!(...) {
    pass!()
}
```

Will remove this macro

Reply:
Some Lisp best practices about macros: 
1. They are meant to reduce cognitive load. Sometimes longer, but simpler code is easier to understand. Sam is going overboard, but I tend to lean in that direction. This is in the middle. It's like the old `try!` macro. But it's not very ergonomic, if we're honest. 
2. Macros need to give you the context of what they do. `propagate` is about the most neutral name a thing can have. It tells me nothing about the context. 
3. It needs to focus the attention of the reader on the key aspects. To me it looks like the error and the validator are the most important. Contrast this with `?`, where it focuses you on the fallible operation. 
4. Macros are more brittle than functions. If you can avoid writing one, use a generic instead (in Lisp, it'd be a regular function). 

For the record, when the `FromResidual` trait is stable, we'd be able to override the behaviour of `?`, so I think we can live with extra code until then.

Reply:
Removed, check out new code